### DISCUSSION TITLE
Support of file:// protocol for Git

### DISCUSSION THREAD

**Discussion by ktomk** [Kong/insomnia#3768] [Ideas] [answered]
I wanted to use the git integration with Insomia, entered a standard file:// git URL and the other data and then started to use it.

With the first push it turned out it did not work as it looks like the underlying Git library / wrapper is not supporting it.

**Comment by nijikokun** [comment]
Welcome @ktomk the library underneath that we use to communicate with Git is `isomorphic-git` it does appear to support the filesystem but would require Insomnia to do detection and a parser for the `file://` as it [only accepts a directory](https://isomorphic-git.org/docs/en/fs) in the documentation and it's not clear if it supports the `file://` url.

That said, what is the purpose of using `file://` what are you looking to get out of it?

---
I've tentatively accepted this as we do indeed want to leverage the file system in some shape for Git sync.

**Comment by ktomk** [comment]
There was no specific preference for the `file://` prefix. I just took `file://` as just entering the absolute path to the Git repository on disk was rejected in the dialogue, prefixing it with `file://` did work (Git support this, too, at least from what I know for local clones).

Regardless, it did let me finish the setup dialogue in Insomnia (Designer), see as well a current screenshot for illustration, it is similar if not identical to the setup dialogue (one difference is that I can't edit the Git URI any longer):

![Selection_129](https://user-images.githubusercontent.com/352517/107274802-33453180-6a51-11eb-9889-b7d95fe016a6.png)

So I'm totally fine if entering the absolute path to the repository would have worked in the first place.

Does this answer your question? @nijikokun

**Comment by ktomk** [comment]


**Reply by wdawson** [reply]
We're starting to revisit sync behavior in general. I think there are many improvements to be made there. I think with that,  it will make sense to take a look at the git initialization behavior too. Don't have any specifics yet but this is on our radar.

**Comment by wdawson** [comment, marked as answer]
Hey, @ktomk ! It looks like this duplicates #3684 so I'm going to be moving this discussion there. As for an update, we're tentatively planning to revisit git sync early next year, the goal being to smooth out the flow in general and we'll be exploring how to have Insomnia act more like a text editor for use-cases like this. Of course, no plan survives contact with the enemy but we know this is one of the big areas to improve on.
